dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestLineInterface.cpp

```diff
@@ -0,0 +1,166 @@
+                                     .setNominalVoltage(360.0)
+                                     .setLowVoltageLimit(340.0)
+                                     .setHighVoltageLimit(420.0)
+                                     .add();
+
+  powsybl::iidm::Bus& vl3Bus1 = vl3.getBusBreakerView().newBus().setId("VL3_BUS1").add();
+
+  powsybl::iidm::VoltageLevel& vl4 = substation2.newVoltageLevel()
+                                     .setId("VL4")
+                                     .setName("VL4_NAME")
+                                     .setTopologyKind(powsybl::iidm::TopologyKind::BUS_BREAKER)
+                                     .setNominalVoltage(225.0)
+                                     .setLowVoltageLimit(200.0)
+                                     .setHighVoltageLimit(260.0)
+                                     .add();
+
+  vl4.getBusBreakerView().newBus().setId("VL4_BUS1").add();
+
+  powsybl::iidm::Line& MyLine = network.newLine()
+                                    .setId("VL1_VL3")
+                                    .setVoltageLevel1(vl1.getId())
+                                    .setBus1(vl1Bus1.getId())
+                                    .setConnectableBus1(vl1Bus1.getId())
+                                    .setVoltageLevel2(vl3.getId())
+                                    .setBus2(vl3Bus1.getId())
+                                    .setConnectableBus2(vl3Bus1.getId())
+                                    .setR(3.0)
+                                    .setX(33.33)
+                                    .setG1(1.0)
+                                    .setB1(0.2)
+                                    .setG2(2.0)
+                                    .setB2(0.4)
+                                    .add();
+
+  LineInterfaceIIDM li(MyLine);
+  ASSERT_EQ(li.getID(), "VL1_VL3");
+  ASSERT_EQ(li.getVNom1(), 380.0);
+  ASSERT_EQ(li.getVNom2(), 360.0);
+  ASSERT_EQ(li.getR(), 3.0);
+  ASSERT_EQ(li.getX(), 33.33);
+  ASSERT_EQ(li.getG1(), 1.0);
+  ASSERT_EQ(li.getG2(), 2.0);
+  ASSERT_EQ(li.getB1(), 0.2);
+  ASSERT_EQ(li.getB2(), 0.4);
+
+  ASSERT_EQ(li.getBusInterface1().get(), nullptr);
+  ASSERT_EQ(li.getBusInterface2().get(), nullptr);
+  const boost::shared_ptr<BusInterface> x_b1(new BusInterfaceIIDM(vl1Bus1));
+  const boost::shared_ptr<BusInterface> x_b3(new BusInterfaceIIDM(vl3Bus1));
+  li.setBusInterface1(x_b1);
+  li.setBusInterface2(x_b3);
+  ASSERT_EQ(li.getBusInterface1().get()->getID(), "VL1_BUS1");
+  ASSERT_EQ(li.getBusInterface2().get()->getID(), "VL3_BUS1");
+
+  MyLine.getTerminal1().disconnect();
+  ASSERT_EQ(li.getVNom1(), 0.0);
+  MyLine.getTerminal2().disconnect();
+  ASSERT_EQ(li.getVNom2(), 0.0);
+  MyLine.getTerminal1().connect();
+  ASSERT_EQ(li.getVNom1(), 380.0);
+  MyLine.getTerminal2().connect();
+  ASSERT_EQ(li.getVNom2(), 360.0);
+
+  ASSERT_TRUE(li.getInitialConnected1());
+  ASSERT_TRUE(li.getInitialConnected2());
```

Comment:
Please add a disconnection and test this value didn't change

Reply:
Please can you confirm in case of disconnection we should keep the same voltage ? That seems strange to me, I would say voltage goes back to 0...

Reply:
done, it works fine